### DISCUSSION TITLE
Is there a way to run android shell commands in termux?

### DISCUSSION THREAD

**Discussion by deoxal** [termux/termux-app#4056] [General]
This command works in adb: getevent -lc4 /dev/input/event0 but I can't use wireless debugging because it gets disabled every time I reboot.

I don't know how well this will work for what I want to do unless I have a wakelock because the battery life on my Pixel 8a is not that great. Is there a way to listen for events such as hardware keypresses without having an endless loop keeping the CPU awake?

**Comment by agnostic-apollo** [comment]
No info on what you are actually trying to listen to and what you intend to do with the events. I suggest you read https://github.com/termux/termux-community/blob/site/site/pages/en/rules/how-to-ask-report-and-request.md

And you likely should be able to use apps like Tasker and/or AutoInput to listen and react to events which use android accessibility service instead of a for loop.

**Comment by deoxal** [comment]
I am trying to listen for hardware keypresess in order to add additional actions when the power or volume buttons are pushed. I don't see why my reason for doing so is necessary to answer my question though.

I have tasker and auto input and I am also trying to do this with that but I do not understand how to create tasks very well and also I encountered a bug that prevented me from doing what I want with Tasker which I contacted the dev about and is helping me with.

All I see for Auto Input is how to detect the coordinates of a touch on the screen not hardware key presses. And on the tasker wiki, it actually said detecting keypresess is not possible but I don't believe that.

> Can Tasker detect volume button presses ?
> No, sorry, that's not possible ATM.
> The closest you can get is to detect changes in a volume variable (e.g. %VOLR for the ringer volume) via event Variable Set.

https://tasker.joaoapps.com/userguide/en/faqs/faq-how.html

**Reply by agnostic-apollo** [reply]
AutoInput has a `AutoInput Key` Tasker Event that can be used to detect volume key presses. You will need to enable its accessibility service in Android Settings and the `Intercept Keys` toggle in AutoInput app itself.
